collated/main/A0153723J.md

```diff
@@ -0,0 +1,318 @@
+# A0153723J
+###### \java\seedu\task\commons\events\ui\DatePickedOnCalendarEvent.java
+``` java
+package seedu.task.commons.events.ui;
+
+import java.time.LocalDate;
+
+import seedu.task.commons.events.BaseEvent;
+
+/**
+ * Event raised when you click a date on the calendar
+ */
+public class DatePickedOnCalendarEvent extends BaseEvent {
+	
+	public final LocalDate date;
+	
+	public DatePickedOnCalendarEvent(LocalDate date) {
+		this.date = date;
+	}
+
+	@Override
+	public String toString() {
+		return this.getClass().getSimpleName();
+	}
+
+}
+```
+###### \java\seedu\task\commons\events\ui\DisplayDirectoryChooserRequestEvent.java
+``` java
+package seedu.task.commons.events.ui;
+
+import seedu.task.commons.events.BaseEvent;
+
+public class DisplayDirectoryChooserRequestEvent extends BaseEvent {
+	
+	public static class DirectoryChooserOperationCancelledException extends Exception {}
+	
+	private String selectedFilePath;
+	
+	public void setSelectedFilePath(String selectedFilePath) {
+		this.selectedFilePath = selectedFilePath;
+	}
+	
+	public String getSelectedFilePath() {
+		return selectedFilePath;
+	}
+
+	@Override
+	public String toString() {
+		return this.getClass().getSimpleName();
+	}
+
+}
+```
+###### \java\seedu\task\logic\commands\StoreCommand.java
+``` java
+package seedu.task.logic.commands;
+
+import java.io.IOException;
+
+import seedu.task.commons.events.ui.DisplayDirectoryChooserRequestEvent.DirectoryChooserOperationCancelledException;
+
+/**
+ * Stores current task book in the specified location.
+ */
+public class StoreCommand extends Command{
+	
+	public static final String COMMAND_WORD = "store";
+	public static final String MESSAGE_USAGE = COMMAND_WORD 
+	        + ": Stores current data in the given location. " 
+		    + "\nParameters: [FILE_LOCATION] (must be an existing directory)" 
+		    + "\nExample: " + COMMAND_WORD + " C:\\Users\\Jim\\Desktop";
+	
+	public static final String MESSAGE_SUCCESS = "Storage Location Updated: %1$s";
+	public static final String MESSAGE_SAVE_CONFIG_FAIL = "Unable to save new location into config";
+	public static final String MESSAGE_CANCEL_STORE_OPERATION = "No file location specified!";
+	
+	private String newSaveLocation;
+	
+	public StoreCommand(String fileLocation) {
+		newSaveLocation = fileLocation;
+	}
+
+	@Override
+	public CommandResult execute() {
+		try {
+			newSaveLocation = model.changeStorageFilePath(newSaveLocation);
+		} catch (DirectoryChooserOperationCancelledException e) {
+			return new CommandResult(MESSAGE_CANCEL_STORE_OPERATION);
+		} catch (IOException e) {
+			return new CommandResult(MESSAGE_SAVE_CONFIG_FAIL);
+		}
+		return new CommandResult(String.format(MESSAGE_SUCCESS, newSaveLocation));
+	}
+
+}
+```
+###### \java\seedu\task\model\ModelManager.java
+``` java
+	@Override
+	public String changeStorageFilePath(String newFilePath) throws DirectoryChooserOperationCancelledException, IOException {
```

Comment:
you seem to be using tabs for indentation? please use 4 spaces for indentation, as per our coding guidelines.
you can get Eclipse to do it for you easily